src/items.ts

```diff
@@ -1,124 +1,1518 @@
+    ITEM_CONTAINERS = "item containers",
+    ORE = "ore",
+    PRODUCT = "product",
+    PURE = "pure",
+    SENSOR = "sensor",
+    STRUCTURAL_PARTS = "structural parts",
 }
 
-export interface Other extends CommonProperties {
-    readonly type: ItemType.OTHER
+/**
+ * Properties commen to each item type
+ */
+export interface CommonProperties {
+    readonly name: string
+    readonly tier: string
+    readonly category: string
+    readonly volume: Liter
 }
 
+/**
+ * Item types
+ */
 export interface Ore extends CommonProperties {
     readonly type: ItemType.ORE
 }
 
-export interface CommonProperties {
-    readonly name: string
-    readonly volume: Liter
+export interface Catalyst extends CommonProperties {
+    readonly type: ItemType.CATALYST
+}
+
+export interface Product extends CommonProperties {
+    readonly type: ItemType.PRODUCT
+}
+
+export interface Element extends CommonProperties {
+    readonly type: ItemType.ELEMENT
+}
+
+/**
+ * Ore type checker
+ * @param item Item to check
+ */
+export function isOre(item: Item): item is Ore {
+    return item.type === ItemType.ORE
 }
 
-export function other(name: string, volume: Liter): Other {
+/**
+ * Returns a new Ore type
+ * @param name Ore name
+ * @param tier Ore tier
+ */
+export function ore(name: string, tier: string): Ore {
     return {
-        type: ItemType.OTHER,
+        type: ItemType.ORE,
         name,
-        volume,
+        tier,
+        category: Category.ORE,
+        volume: 1,
     }
 }
 
-export function machine(name: string, volume: Liter): Machine {
+/**
+ * Returns a new Catalyst type
+ * @param name Catalyst name
+ * @param tier Catalyst tier
+ * @param volume Catalyst volume
+ */
+export function catalyst(name: string, tier: string, volume: Liter): Catalyst {
     return {
-        type: ItemType.MACHINE,
+        type: ItemType.CATALYST,
         name,
+        tier,
+        category: Category.CATALYST,
         volume,
     }
 }
 
-export function isMachine(item: Item): item is Machine {
-    return item.type === ItemType.MACHINE
-}
-
-export function container(name: string, volume: Liter): Container {
+/**
+ * Returns a new Product type
+ * @param name Product name
+ * @param tier Product tier
+ * @param category Product category
+ * @param volume Product volume
+ */
+export function product(name: string, tier: string, category: string, volume: Liter): Product {
     return {
-        type: ItemType.CONTAINER,
+        type: ItemType.PRODUCT,
         name,
+        tier,
+        category,
         volume,
     }
 }
 
-export function isContainer(item: Item): item is Container {
-    return item.type === ItemType.CONTAINER
-}
-
-export function ore(name: string): Ore {
+/**
+ * Returns a new Element type
+ * @param name Element name
+ * @param tier Element tier
+ * @param category Element category
+ * @param volume Element volume
+ */
+export function element(name: string, category: string, volume: Liter): Element {
```

Comment:
```suggestion
export function element(name: string, category: Category, volume: Liter): Element {
```

Reply:
Fixed in https://github.com/lgfrbcsgo/du-factory-generator/commit/9413cb40975215110ead9dad189d664c82f8b980